lisa/analysis/rta.py

```diff
@@ -15,187 +15,493 @@
+
+        return (start_time, end_time)
+
+    PhaseWindow = namedtuple("PhaseWindow", ['id', 'start', 'end'])
+
+    @df_rtapp_phases_start.used_events
+    def task_phase_window(self, task, phase):
+        """
+        Return the window of a requested task phase.
+
+        For the specified ``task`` it returns a tuple with the (start, end)
+        time of the requested ``phase``. A negative phase number can be used to
+        count phases backward from the last (-1) toward the first.
+
+        :param task: Task name or PID to look for
+        :type task: int or str
+
+        :param phase: The ID of the phase to consider
+        :type phase: int
+
+        :returns: tuple(start_s, end_s)
+        """
+        phase_start = self.df_rtapp_phase_start(task, phase)
+        phase_end = self.df_rtapp_phase_end(task, phase)
+
+        return self.PhaseWindow(phase, phase_start, phase_end)
+
+    ###########################################################################
+    # rtapp_phase events related methods
+    ###########################################################################
+
+    @requires_events('rtapp_event')
+    def df_rtapp_event(self, task=None):
+        df = self.trace.df_events('rtapp_event')
+        return self._task_filtered(df, task)
+
+    @df_rtapp_event.used_events
+    def df_rtapp_task_activations(self, task):
+        """
+        Dataframe of all activations of the specified task
+
+        :param task: the actual trace task name for an rt-app task to consider
+        :type task: str
+        """
+        df = self.df_rtapp_event(task)
+        return (df[(df["__comm"] == task) & (df.desc == "run")]
+                [["__cpu", "__pid"]])
+
+    ###########################################################################
+    # rtapp_stats events related methods
+    ###########################################################################
+
+    @memoized
+    @requires_events('rtapp_stats')
+    def _get_stats(self):
+        df = self.trace.df_events('rtapp_stats')
         # Add performance metrics column, performance is defined as:
         #             slack
         #   perf = -------------
         #          period - run
-        df['PerfIndex'] = df['Slack'] / (df['CPeriod'] - df['CRun'])
+        df.loc[:,'perf_index'] = df['slack'] / (df['c_period'] - df['c_run'])
 
         return df
 
-    @property
-    def tasks(self):
-        """
-        List of tasks for which performance data have been loaded
+    @_get_stats.used_events
+    def df_rtapp_stats(self, task=None):
+        df = self._get_stats()
+        return self._task_filtered(df, task)
+
+    @memoized
+    @df_rtapp_stats.used_events
+    def df_phases(self, task):
         """
-        return sorted(self.perf_data.keys())
+        Get phases actual start times and durations
+
+        :param task: Name of the task that we want the phases dataframe of
+        :type task: def __str__(self):
```

Comment:
`s/def __str__(self):/str/`

Reply:
WTF... spacemacs autocompletion fires when it should now :)